meteor/server/databaseMigrationSystemSteps.ts

```diff
@@ -102,6 +102,127 @@ function ensureCollectionProperty<T = any> (
+		}
+		allDevices[playoutGw._id] = devices
+	})
+
+	// iterate over devices for gateways
+	_.each(allDevices, (devices, key) => {
+		allDevices[key] = _.pick(devices, (device) => {
+			let needsMoreData = false
+			let foundHostProp = objectPathGet(device, 'options.host', undefined) 
+			let foundPortProp = objectPathGet(device, 'options.port', undefined) 
+			// let foundHostProp = objectPathGet(device, 'options.host', '') // @ TODOD: discuss if host should be required to be initiated as empty string
+			if (foundHostProp !== undefined && typeof foundHostProp === 'string') {
+				// if host is string but length is 0, then add to mossingHostDevices
+				if(foundHostProp.trim().length === 0){
+					needsMoreData = true
+				}
+			}
+
+			// port not number or <= 0
+			if(foundPortProp !== undefined && (isNaN(foundPortProp) || foundPortProp <= 0)) {
+				needsMoreData = true
+			}
+			
+			return needsMoreData
+		})
+	})	
+
+	// create migrationsteps for all affected devices
+	return {
+		id: `ensureHostPortOnDevicesOfType${deviceType > -1 ? deviceType : 'All'}`,
+		canBeRunAutomatically: false,
+		validate: () => {
+			for (let key in allDevices) {
+				if (!_.isEmpty(allDevices[key])) {
+					return 'Some playout devices are missing host or port'
+				}
+			}
+			return true
+		},
+		input: () => {
+			// let objects = collection.find(selector).fetch()
+			let inputs: Array<MigrationStepInput> = []
+			
+			_.each(allDevices, (devices: {[key:string]: PlayoutDeviceSettingsDevice}, gatewayId: string) => {
+				_.each(devices, (device: PlayoutDeviceSettingsDevice, deviceId: string) => {
+
+					// should have host added
+					let foundHostProp = objectPathGet(device, 'options.host', undefined) 
+					// let foundHostProp = objectPathGet(device, 'options.host', '') // @ TODOD: discuss if host should be required to be initiated as empty string
+					if (foundHostProp !== undefined && typeof foundHostProp === 'string') {
+						// if host is string but length is 0, then add to mossingHostDevices
+						if(foundHostProp.trim().length === 0){
+							inputs.push({
+								label: `Playout device "${deviceId}" on gateway "${gatewayId}" is missing host`,
+								description: `Host for device "${deviceId}"`,
+								inputType: 'text',
+								attribute: `${gatewayId}.${deviceId}.options.host`,
+								defaultValue: null
+							})
+						}
+					}
+					
+					// should have port added
+					let foundPortProp = objectPathGet(device, 'options.port', undefined) 
+					// port not number or <= 0
+					if(foundPortProp !== undefined && (isNaN(foundPortProp) || foundPortProp <= 0)) {
+						inputs.push({
+							label: `Playout device "${deviceId}" on gateway "${gatewayId}" is missing port`,
+							description: `Port for device "${deviceId}"`,
+							inputType: 'int',
+							attribute: `${gatewayId}.${deviceId}.options.port`,
+							defaultValue: DefaultPlayoutDevicePort[device.type] || 0
```

Comment:
@nytamin see asana task on default values not working (never making it to the migration)